var/spack/repos/builtin/packages/py-region-grower/package.py

```diff
@@ -13,23 +13,27 @@ class PyRegionGrower(PythonPackage):
     git      = "git@bbpgitlab.epfl.ch:neuromath/region-grower.git"
 
     version('develop', branch='master')
-    version('0.2.3', tag='region-grower-v0.2.3')
-    version('0.1.10', tag='region-grower-v0.1.10')
-    version('0.1.5', tag='region-grower-v0.1.5')
-    version('0.1.2', tag='region-grower-v0.1.2')
+    version('0.3.0', tag='region-grower-v0.3.0')
+    version('0.2.4', tag='region-grower-v0.2.4')
 
     depends_on('py-setuptools', type='build')
 
     depends_on('py-click@7.0:', type='run')
     depends_on('py-tqdm@4.28.1:', type='run')
-    depends_on('py-tns@2.4.4:', type='run')
-    depends_on('py-voxcell@2.7:', type='run')
-    depends_on('py-diameter-synthesis@0.2.4:', type='run')
-    depends_on('py-morphio@2.7:', type='run')
-    depends_on('py-neuroc@0.2.7:', type='run')
-    depends_on('py-neurom@2.2.1:2.999', type='run')
-    depends_on('py-morph-tool@2.5.0:', type='run')
+    depends_on('py-tns@2.5.0:', type='run', when='@0.3.0:')
+    depends_on('py-tns@:2.4.99', type='run', when='@:0.2.99')
+    depends_on('py-voxcell@2.7:3.999', type='run')
+    depends_on('py-diameter-synthesis@0.2.5:', type='run', when='@0.3.0:')
+    depends_on('py-diameter-synthesis@:0.2.4', type='run', when='@:0.2.99')
+    depends_on('py-morphio@3.0:3.999', type='run', when='@0.3.0:')
+    depends_on('py-morphio@2.7:2.99', type='run', when='@:0.2.99')
+    depends_on('py-neuroc@0.2.8:', type='run', when='@0.3.0:')
+    depends_on('py-neuroc@:0.2.7', type='run', when='@:0.2.99')
+    depends_on('py-neurom@3.0:3.999', type='run', when='@0.3.0:')
+    depends_on('py-neurom@2.2.1:2.99', type='run', when='@:0.2.99')
+    depends_on('py-morph-tool@2.8.0:', type='run', when='@0.3.0:')
+    depends_on('py-morph-tool@:2.7.99', type='run', when='@:0.2.99')
     depends_on('py-attrs@19.3.0:', type='run')
-    depends_on('py-dask+dataframe+distributed@2.15.0:', type='run')
-    depends_on('py-dask-mpi@2.21.0:', type='run')
-    depends_on('py-mpi4py@3.0.3:', type='run')
+    depends_on('py-dask+dataframe+distributed@2.15.0:2.30.0', type='run')
+    depends_on('py-dask-mpi@:2.21.0', type='run')
+    depends_on('py-mpi4py@:3.0.3', type='run')
```

Comment:
I wanted to restrict to the versions that are available on our spack.

Reply:
I think the last changes are ok, but in setup.py there is `morph-tool[nrn]>=2.9,<3` while in spack it's >= 2.8.0 (and 2.9.0 is added in this PR)

Reply:
For the `dask` requirement in region-grower, there is a fix in the master branch of `dask` so the next release should work, thus only these two versions do not work. But I didn't know if the next version would be `2021.8.3` or `2021.9.1`. Now that we are in September, I guess it will be `2021.9.1` :)

Reply:
For the records, this is the dask PR that solved the issue: https://github.com/dask/distributed/pull/5257